src/actions/organization.tsx

```diff
@@ -0,0 +1,127 @@
+import { ActionCreatorsMapObject, Dispatch, Action } from 'redux'
+import { IOrganization } from '../reducers/organization'
+
+// API Helpers
+import { POST } from '../helpers/api'
+import { PUT } from '../helpers/api'
+import { GET } from '../helpers/api'
+import { DELETE } from '../helpers/api'
+
+export const FETCHED_ORGANIZATIONS = 'FETCH_ORGANIZATIONS'
+export const CREATE_ORGANIZATION = 'CREATE_ORGANIZATION'
+export const CREATED_ORGANIZATION = 'CREATED_ORGANIZATION'
+
+export const DELETE_ORGANIZATION = 'DELETE_ORGANIZATION'
+export const DELETED_ORGANIZATION = 'DELETE_ORGANIZATION'
+
+// -----------------------------------------------------------------------------
+// Get all Organizations
+// -----------------------------------------------------------------------------
+/**
+ * Gets all organizations via a call to the `/organizations` API.
+ */
+const fetchOrganizations = () => (dispatch: Dispatch<Action>) => {
+	return GET('/organizations', {}).then((organizations: IOrganization[]) => {
+		dispatch(fetchedOrganizations(organizations))
+	}).catch((err) => Promise.reject(err))
+}
+/**
+ * Builds a `IFetchedOrganizationsAction ` upon successful fetch.
+ * @param organizations The response from the organizations API call.
+ * @return The `FetchedOrganizationsAction` instance.
+ */
+const fetchedOrganizations = (organizations: IOrganization[]): Action => {
+	const action: IFetchedOrganizationsAction = {
+		organizations: organizations,
+		receivedAt: Date.now(),
+		type: FETCHED_ORGANIZATIONS,
+	}
+	return action
+}
+/**
+ * IFetchedOrganizationsAction is dispatched after organizations are fethed.
+ */
+export interface IFetchedOrganizationsAction extends Action {
+	type: string,
+	organizations: IOrganization[],
+	receivedAt: number
+}
+// -----------------------------------------------------------------------------
+// Create or update organization depending on button type
+// -----------------------------------------------------------------------------
+/**
+ * Creates a new organization via a call to the `/organizations` API.
+ * @param name The name supplied in the organization form.
+ * @param id Auto-assigned id number.
+ */
+const createOrganization = ( id: number, name: string ) => (dispatch: Dispatch<ICreatedOrganizationAction>) => { 
```

Comment:
We want to have separate functions for creating and updating a functions. So we want to have `createOrganizations` and `updateOrganization`.